mmengine/_strategy/deepspeed.py

```diff
@@ -517,35 +521,56 @@ def save_checkpoint(
             extra_ckpt (dict, optional): Extra checkpoint to save.
                 Defaults to None.
         """
-        if extra_ckpt is None:
-            extra_ckpt = dict()
-        if 'meta' not in extra_ckpt:
-            extra_ckpt['meta'] = dict()
-        extra_ckpt['meta'].update(
-            seed=self.seed,
-            time=time.strftime('%Y%m%d_%H%M%S', time.localtime()),
-            mmengine=mmengine.__version__ + get_git_hash(),
-        )
-
-        if save_optimizer and hasattr(self, 'optim_wrapper'):
-            # The key can not be 'optimizer', otherwise error will be thrown
-            # when loading or resuming checkpoint.
-            extra_ckpt['optim_wrapper'] = self.optim_state_dict()
-
-        if save_param_scheduler and hasattr(self, 'param_schedulers'):
-            extra_ckpt['param_schedulers'] = self.scheduler_state_dict()
+        if (not save_optimizer
+                and self.config.zero_optimization.get('stage', 0) == 3
+                and not self.config.zero_optimization.get(
+                    'stage3_gather_16bit_weights_on_model_save', False)):
```

Comment:
- `zero_optimization` is not necessarily in `self.config`
- `stage3_gather_16bit_weights_on_model_save` will be deprecated, so it is recommended to use `gather_16bit_weights_on_model_save`. https://deepspeed.readthedocs.io/en/stable/zero3.html#deepspeed.runtime.zero.config.DeepSpeedZeroConfig.gather_16bit_weights_on_model_save